[14.0] DES and EMEBI : update, improve and simplify

alexis-via opened the pull request:
l10n_fr_intrastat_product (EMEBI):
- adapt to the very big changes I made in intrastat_product
- DEB -> EMEBI
- remove intrastat_fiscal_representative_id on res.partner
- add a default intrastat transaction for B2C
- filter intrastat_transaction_id on invoice form depending on move_type

l10n_fr_intrastat_service (DES):
- auto-generate XML export upon confirmation to harmonize ergonomy with intrastat product
- auto-delete XML upon back2draft
- add confirmation pop-up when going back to draft
- fix bug on total amount and number of lines when removing all lines
- adapt for the fact that EU B2C fiscal positions now have intrastat=True

Depends on PR https://github.com/OCA/intrastat-extrastat/pull/198

alexis-via committed:
l10n_fr_intrastat_* : update, improve and simplify

l10n_fr_intrastat_product:
- adapt to the very big changes I made in intrastat_product
- DEB -> EMEBI
- remove intrastat_fiscal_representative_id on res.partner
- add a default intrastat transaction for B2C
- filter intrastat_transaction_id on invoice form depending on move_type

l10n_fr_intrastat_service:
- auto-generate XML export upon confirmation to harmonize ergonomy with intrastat product
- auto-delete XML upon back2draft
- add confirmation pop-up when going back to draft
- fix bug on total amount and number of lines when removing all lines
- adapt for the fact that EU B2C fiscal positions now have intrastat=True

alexis-via commented:
One important note: don't try to upgrade from previous version with existing data, it won't work ! Writing migration scripts will be a big task... but I guess we'll have to do it one day.

legalsylvain commented:
hi @alexis-via. thanks for this huge contribution.
> One important note: don't try to upgrade from previous version with existing data, it won't work ! Writing migration scripts will be a big task... but I guess we'll have to do it one day.

Not sure to understand. Did you plan to merge this PR before providing migration scripts ? Or this PR will stay opened ?

alexis-via commented:
@legalsylvain As explained on the underlying PR on OCA/intrastat-extrastat, I don't think the underlying PR will be merged. Maybe I will have some remarks that will lead to more changes in the datamodel, so I'd better wait for some first feedback before working on the migration script to avoid double work. Anyway, I'll have to write migration script to migrate my own customers, so I'll have to write them one day, and it will be a big work (and I'm not an expert in this field, so it takes me quite some time for me).

But I wanted to publish this PR now, to get some feedback on the changes I propose and also to allow new installations to have a working version for DEB/EMEBI (the previous version was broken following a big Brexit PR that was over-complex https://github.com/OCA/intrastat-extrastat/pull/163)

legalsylvain commented:
Thanks a lot for your explanation. perfectly clear.

alexis-via committed:
l10n_fr_intrastat_service: fix mail.template for reminder

alexis-via committed:
emebi: adapt to new warn msg code

auto-reformatting

alexis-via committed:
emebi: Adapt post-install script

Set company_id to False on intrastat.transaction

alexis-via committed:
DES: fix and improve tests

Code cleanup.

alexis-via committed:
l10n_fr_intrastat_product: call super on _generate_xml

_generate_xml() : Don't raise if company_id.fr_intrastat_accreditation
is not set, to avoid blocking draft->done

fmdl commented on l10n_fr_intrastat_product/models/intrastat_product_declaration.py:
```diff
@@ -95,103 +78,32 @@ def _get_fr_department(self, inv_line, notedict):
             if so_line:
                 so = so_line.order_id
-                dpt = so.warehouse_id.get_fr_department()
+                dpt = so.warehouse_id._get_fr_department()
```
le champ warehouse_id est présent sur les so_lines
`dpt = so_line.warehouse_id._get_fr_department()`

j'ai un client qui fait du multi entrepot par commande.

alexis-via committed:
l10n_fr_intrastat_product: Improve migration script

Update data for account.fiscal.position.template
